### PR TITLE
DOC: improve fillna() doc for limit keyword (fixes #10002)

### LINKED ISSUE #10002: Usage of 'limit' keyword when filling with a value
From SO: http://stackoverflow.com/questions/29906512/pandas-variation-on-filling-nan-whats-an-elegant-and-pythonic-way-to-do-thi/29906743?noredirect=1#29906743

When doing `df.fillna(10, limit=2)` what is `limit=2` doing?

The [docstring](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.fillna.html) says:

> limit : int, default None
>               Maximum size gap to forward or backward fill

But this only explains for filling with a method, not for filling with a value.

### PR DESCRIPTION
**Pull request by mortada**
fixes #10002

### CODE REVIEW

File: pandas/core/categorical.py [hunk 1]
```diff
@@ -1180,7 +1180,9 @@ def fillna(self, fill_value=None, method=None, limit=None):
         value : scalar
             Value to use to fill holes (e.g. 0)
         limit : int, default None
-            Maximum size gap to forward or backward fill (not implemented yet!)
+            If method is specified, this is the maximum size gap to forward or backward
+            fill. Otherwise this is the maximum number of entries along the axis where
+            holes will be filled. (not implemented yet!)
```
Comment (jorisvandenbossche) [inline pandas/core/categorical.py]: Maybe put the not implemented yet first? (and maybe also explicitely "for Categorical")

**Comment by jorisvandenbossche** [comment]
Actually, in the way I interpret "maximum size gap to forward or backward fill", this is not fully correct.

I interpret it as "only fill that gap if the gap is not larger as 'limit'", but it is "per gap, fill a maximum numbers of 'limit'":

```
In [10]: s=pd.Series(range(4), index=[1,2,5,9])

In [11]: s = s.reindex(range(10))

In [12]: s
Out[12]:
0   NaN
1     0
2     1
3   NaN
4   NaN
5     2
6   NaN
7   NaN
8   NaN
9     3
dtype: float64

In [13]: s.fillna(method='ffill', limit=2)
Out[13]:
0   NaN
1     0
2     1
3     1
4     1
5     2
6     2
7     2
8   NaN
9     3
dtype: float64
```

File: pandas/core/categorical.py (hunk 1 above)
Comment (mortada) [inline pandas/core/categorical.py]: good idea, will update

**Comment by mortada** [comment]
What if we rephrase it to say something like:

"maximum number of consecutive holes to forward or backward fill"

**Comment by jreback** [comment]
lgtm

**Comment by jorisvandenbossche** [comment]
It is a bit subjective, but I would personally still interpret that not fully correct (I see a hole as possibly multiple NaN values if they are consecutive, but maybe this is not correct language?). 
Otherwise: "maximum number of consecutive NaN values to forward/backward file per hole" 

I think an example in the docstring to show this behaviour would possibly be the most clarifying.

**Comment by mortada** [comment]
Ah yeah I suppose you could interpret a "hole" as multiple NaNs. @jorisvandenbossche please see the updated changes in the diff, I wonder if it's now clear to you.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -2319,7 +2319,11 @@ def fillna(self, value=None, method=None, axis=None, inplace=False,
             other views on this object, (e.g. a no-copy slice for a column in a
             DataFrame).
         limit : int, default None
-            Maximum size gap to forward or backward fill
+            If method is specified, this is the maximum number of consecutive
+            NaN values to forward/backward fill. In other words, if there is
+            a gap with more than this number of consecutive NaNs, it will only
+            be partially filled. If method is not specified, this is the
+            maximum number of entries along the axis where NaNs will be filled.
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: one small nitpick: maybe "along the *entire* axis"

For the rest everything clear now! Thanks!

Comment (mortada) [inline pandas/core/generic.py]: sure definitely more clear now, thanks!

**Comment by jorisvandenbossche** [comment]
Thanks!
